### ISSUE TITLE
ps.map() only calls mapping function at most 16 times

### ISSUE THREAD

**Issue by aoberoi** [spion/promise-streams#6]
i've observed that `MapPromiseStream`'s only call their `fn` argument at most 16 times. this means that when a readable stream that emits more than 16 `'data'` events is piped to it, `fn` never gets a chance to run. moreover, the promise returned by `ps.wait(mps)` never resolves or rejects when `mps` is one suffering from the previous problem.

i've reproduced the issue in a branch on my fork that can be seen here: aoberoi@2f651d1144b589. it isn't a proper test but its useful to illustrate the problem. the output is shown below.

node version: ```v4.1.1```

output:
```
TAP version 13
# ps.wait
ok 1 should wait for the last element
# map-wait
ok 2 should wait for the last element
# map-wait 2
reading 1
reading 2
reading 3
reading 4
reading 5
reading 6
reading 7
reading 8
reading 9
reading 10
reading 11
reading 12
reading 13
reading 14
reading 15
reading 16
reading 17
reading 18
reading 19
reading 20
reading 21
reading 22
reading 23
reading 24
reading 25
reading 26
reading 27
reading 28
reading 29
reading 30
reading 31
reading 32
doubling 1
doubling 2
doubling 3
doubling 4
doubling 5
doubling 6
doubling 7
doubling 8
doubling 9
doubling 10
doubling 11
doubling 12
doubling 13
doubling 14
doubling 15
doubling 16
reading 33
reading 34
reading 35
reading 36
reading 37
reading 38
reading 39
reading 40
reading 41
reading 42
reading 43
reading 44
reading 45
reading 46
reading 47
reading 48
not ok 3 test exited without ending
  ---
    operator: fail
  ...
not ok 4 test exited without ending
  ---
    operator: fail
  ...
not ok 5 test exited without ending
  ---
    operator: fail
  ...

1..5
# tests 5
# pass  2
# fail  3
```

you can see that the mapper (outputs `doubling x`) is only called 16 times, and the readable stream doesn't even finish since its only being read 48 times.

**Comment by spion** [comment]
Yeah, it seems like the new stream updates (node 0.12 and 4.x) broke promise-streams. Sigh. Trying to figure it out.

**Comment by spion** [comment]
Fixed in v1.0.1 - I bumped the major version because this may be incompatible with the previous streams2 behaviour. This version will be safe to upgrade to in node 0.12 and up, but may cause breakage if used in node v0.10. The old version will continue to work fine in node v0.10

**Comment by aoberoi** [comment]
@spion Thanks for getting this out so fast! superb!
